stackit/stackit_core.py

```diff
@@ -39,24 +39,31 @@ def focusQuestion(questions, count):
     userInput = '0'
     #Looping while the user wants to see more input
     while(userInput != 'm'):
-        userInput = promptUser("Press m for more, or a number to select: ")
-        if(userInput == 'm'):
-            break
-        if(0 < int(userInput) and int(userInput) <= count):
-            print("\n\n\n\n\n\n")
-            printFullQuestion(questions[int(userInput)- 1])
-            branchInput = '0'   #user deciding whether to branch to open browser, or to return to search
-            while(branchInput != 'x'):
-                branchInput = promptUser("Press b to launch browser, or x to return to search: ")
-                if(branchInput == 'b'):
-                    webbrowser.open(questions[int(userInput)-1].json['link'], new=0, autoraise=True)
-            #User selects x to return to search
-            if(branchInput == 'x'):
-                print("\n\n\n\n\n\n\n\n\n\n\n\n")
-                #Ranging over the 5 questions including the user's choice
-                for j in range(5*int((int(userInput)-1)/5), 5*int((int(userInput)-1)/5)+5):
-                    printQuestion(questions[j], j+1)
-                continue   #exit the inner while loop
+        userInput = promptUser("Enter m for more, a question number to select, or q to exit: ")
+        try:
+            if(userInput == 'q'):
+                sys.exit()
+            if(userInput == 'm'):
+                break
+            if(0 < int(userInput) and int(userInput) <= count):
+                print("\n\n\n\n\n\n")
+                printFullQuestion(questions[int(userInput)- 1])
+                branchInput = '0'   #user deciding whether to branch to open browser, or to return to search
+                while(branchInput != 'x'):
+                    branchInput = promptUser("Enter b to launch browser, or x to return to search: ")
+                    if(branchInput == 'b'):
+                        webbrowser.open(questions[int(userInput)-1].json['link'], new=0, autoraise=True)
+                #User selects x to return to search
+                if(branchInput == 'x'):
+                    print("\n\n\n\n\n\n\n\n\n\n\n\n")
+                    #Ranging over the 5 questions including the user's choice
+                    for j in range(5*int((int(userInput)-1)/5), 5*int((int(userInput)-1)/5)+5):
+                        printQuestion(questions[j], j+1)
+                    continue   #exit the inner while loop
+        except:
+            if (userInput != 'q'):
+                print("The input entered was not recognized as a valid choice.  Exiting...")
+            sys.exit()
```

Comment:
I think I'd rather it was `continue` than `sys.exit()` in the case that
some invalid character is entered. e.g.:

    except:
        if (userInput != 'q'):
            print("The input entered was not recognizes as a valid choice.")
            continue
        else: # If the user input is 'q', no input
            sys.exit()

On Wed, Feb 4, 2015 at 12:55 PM, Phil <notifications@github.com> wrote:

> In stackit/stackit_core.py
> <https://github.com/lukasschwab/stackit/pull/4#discussion_r24118595>:
>
> > +                branchInput = '0'   #user deciding whether to branch to open browser, or to return to search
> > +                while(branchInput != 'x'):
> > +                    branchInput = promptUser("Enter b to launch browser, or x to return to search: ")
> > +                    if(branchInput == 'b'):
> > +                        webbrowser.open(questions[int(userInput)-1].json['link'], new=0, autoraise=True)
> > +                #User selects x to return to search
> > +                if(branchInput == 'x'):
> > +                    print("\n\n\n\n\n\n\n\n\n\n\n\n")
> > +                    #Ranging over the 5 questions including the user's choice
> > +                    for j in range(5*int((int(userInput)-1)/5), 5*int((int(userInput)-1)/5)+5):
> > +                        printQuestion(questions[j], j+1)
> > +                    continue   #exit the inner while loop
> > +        except:
> > +            if (userInput != 'q'):
> > +                print("The input entered was not recognized as a valid choice.  Exiting...")
> > +            sys.exit()
>
> So would you rather it be continue then sys.exit()?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lukasschwab/stackit/pull/4/files#r24118595>.
>



-- 
Lukas Schwab